R/unneeded_concatenation_linter.R

```diff
@@ -5,7 +5,10 @@
 #' @param allow_single_expression Logical, default `TRUE`. If `FALSE`, one-expression
 #'   usages of `c()` are always linted, e.g. `c(x)` and `c(matrix(...))`. In some such
 #'   cases, `c()` is being used for its side-effect of stripping non-name attributes;
-#'   it is usually preferable to use [as.vector()] to accomplish the same more readably.
+#'   it is usually preferable to use the more readable [as.vector()] instead. But the
```

Comment:
Suggestion:

`as.vector()` is not always preferable, for example with environments (especially, R6 objects), in which case `list()` is the better alternative.

(note that what you've said is not exactly right -- `c(r6_object)` doesn't do any conversion, it just plops the object into a list, like `list(r6_object)` is doing, i.e., `identical(r6_object, c(r6_object)[[1]])`)

Reply:
Thanks for the clarification.

I am using the suggested phrasing now.